src/internal/m365/graph/errors.go

```diff
@@ -267,6 +279,12 @@ func IsErrSiteNotFound(err error) bool {
 	return hasErrorMessage(err, requestedSiteCouldNotBeFound)
 }
 
+func IsErrResourceLocked(err error) bool {
+	return errors.Is(err, ErrResourceLocked) ||
+		hasInnerErrorCode(err, ResourceLocked) ||
+		hasErrorCode(err, NotAllowed)
```

Comment:
It's what graph gives us. See the following response body:
```json
{
    "error": {
        "code": "notAllowed",
        "message": "Access to this site has been blocked. Please contact the administrator to resolve this problem.",
        "innerError": {
            "code": "resourceLocked",
            "date": "2023-10-10T17:08:49",
            "request-id": "f418b448-042c-4ad8-a938-c6494e600bb4",
            "client-request-id": "f418b448-042c-4ad8-a938-c6494e600bb4"
        }
    }
}
```

This seems like an easy catch, but the graph api client interface doesn't expose `code` as a property on innerError.  We're looking at the "additionalData" for it, which might still catch it.  But otherwise we're left with the primary error code as the identifier.

I'll double check that we don't get this same code from permissions access issues.  If it overlaps I'll remove it.  But otherwise I think it's fine to keep in place.

Reply:
Thanks for clarifying! The plan sounds good to me.

Reply:
yup, permission errors give `Authorization_RequestDenied` as the code.